common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -221,6 +231,38 @@ def perform_destroy(self, instance):
         # This is not called with is_refund=True here because it is assumed the user has already been refunded.
         _process_revoke_and_unenroll_entitlement(instance)
 
+    def partial_update(self, request, *args, **kwargs):
+        entitlement_uuid = kwargs.get('uuid', None)
+
+        try:
+            entitlement = CourseEntitlement.objects.get(uuid=entitlement_uuid)
+        except CourseEntitlement.DoesNotExist:
+            return HttpResponseBadRequest(
+                u'Could not find entitlement {entitlement_uuid} to update'.format(
+                    entitlement_uuid=entitlement_uuid
+                )
+            )
+        support_details = request.data.pop('support_details', [])
+
+        for support_detail in support_details:
+            support_detail['entitlement'] = entitlement
+            support_detail['support_user'] = request.user
+            unenrolled_run_id = support_detail.get('unenrolled_run', None)
+            if unenrolled_run_id:
+                try:
+                    with transaction.atomic():
+                        unenrolled_run_course_key = CourseKey.from_string(unenrolled_run_id)
+                        _unenroll_entitlement(entitlement, unenrolled_run_course_key)
+                        support_detail['unenrolled_run'] = CourseOverview.objects.get(id=unenrolled_run_course_key)
+                except Exception as error:  # pylint: disable=broad-except
```

Comment:
We should avoid adding new Broad exceptions.  Please look at this for the actual exceptions we are concerned about.